pkg/security/grantvmgroupaccess.go

```diff
@@ -0,0 +1,285 @@
+	accessMaskStandardRightsRequired accessMask = 0x000F0000
+	accessMaskStandardRightsAll      accessMask = 0x001F0000
+	accessMaskSpecificRightsAll      accessMask = 0x0000FFFF
+
+	accessMaskDesiredPermission accessMask = 0x12019f // Hard-coded version of what we need to set
+
+	accessModeNotUsed         accessMode = 0
+	accessModeGrant           accessMode = 1
+	accessModeSet             accessMode = 2
+	accessModeDeny            accessMode = 3
+	accessModeRevoke          accessMode = 4
+	accessModeSetAudit        accessMode = 5
+	accessModeSetAuditFailure accessMode = 6
+
+	desiredAccessDelete      desiredAccess = 0x10000
+	desiredAccessReadControl desiredAccess = 0x20000
+	desiredAccessWriteDac    desiredAccess = 0x40000
+	desiredAccessWriteOwner  desiredAccess = 0x80000
+	desiredAccessSynchronize desiredAccess = 0x100000
+
+	gvmga = "GrantVmGroupAccess:"
+
+	inheritModeNoInheritance                  inheritMode = 0x0
+	inheritModeSubObjectsOnlyInherit          inheritMode = 0x1
+	inheritModeSubContainersOnlyInherit       inheritMode = 0x2
+	inheritModeSubContainersAndObjectsInherit inheritMode = 0x3
+	inheritModeNoPropagate                    inheritMode = 0x4
+	inheritModeInheritOnly                    inheritMode = 0x8
+	inheritModeInheritedAce                   inheritMode = 0x10
+
+	objectTypeFileObject objectType = 0x1
+
+	securityInformationOwner securityInformation = 0x1
+	securityInformationGroup securityInformation = 0x2
+	securityInformationDACL  securityInformation = 0x4
+	securityInformationSacl  securityInformation = 0x8
+	securityInformationLabel securityInformation = 0x10
+
+	shareModeRead   shareMode = 0x1
+	shareModeWrite  shareMode = 0x2
+	shareModeDelete shareMode = 0x4
+
+	sidVmGroup                   = "S-1-5-83-1-3166535780-1122986932-343720105-43916321"
+	sidVmWorkerProcessCapability = "S-1-15-3-1024-2268835264-3721307629-241982045-173645152-1490879176-104643441-2915960892-1612460704"
+
+	trusteeFormIsSid            trusteeForm = 0
+	trusteeFormIsName           trusteeForm = 1
+	trusteeFormBadForm          trusteeForm = 2
+	trusteeFormIsObjectsAndSid  trusteeForm = 3
+	trusteeFormIsObjectsAndName trusteeForm = 4
+
+	trusteeTypeUnknown        trusteeType = 0
+	trusteeTypeUser           trusteeType = 1
+	trusteeTypeGroup          trusteeType = 2
+	trusteeTypeDomain         trusteeType = 3
+	trusteeTypeAlias          trusteeType = 4
+	trusteeTypeWellKnownGroup trusteeType = 5
+	trusteeTypeDeleted        trusteeType = 6
+	trusteeTypeInvalid        trusteeType = 7
+	trusteeTypeComputer       trusteeType = 8
+)
+
+// GrantVMGroupAccess sets the DACL for the group SIDs to the specified
+// file or directory. This is a golang re-implementation of the same function
+// in vmcompute, just not exported in RS5. Which kind of sucks. Sucks a lot :/
+func GrantVmGroupAccess(name string) error {
+
+	// Stat (we use this to determine if file/directory is a directory).
+	s, err := os.Stat(name)
```

Comment:
Should we verify `name` ends with `vhd` or `vhdx`?

Reply:
Why make this library function less generic? I don't think we should.

Reply:
Because we need it for exactly 1 thing. To set the acl on vhd files. When do we need a folder and all children?

Reply:
You asked me to implement GrantVmGroupAccess in go, which is exactly what I did :smile:. How we use it is separate from what the C implementation does.... Who knows whether directory support might be needed in the future?